File: Selenium/Project/PActivity9.java
package Project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;

import java.time.Duration;
import java.util.List;

public class PActivity9 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        Actions builder = new Actions(driver);
        driver.get("http://alchemy.hguy.co/crm");
        // Login
        driver.findElement(By.name("user_name")).sendKeys("admin");
        driver.findElement(By.name("username_password")).sendKeys("pa$$w0rd");
        driver.findElement(By.id("bigbutton")).click();
        // Navigate to Sales -> Leads
        driver.findElement(By.id("grouptab_0")).click();
        WebElement Leads = driver.findElement(By.cssSelector("ul.navbar-nav li.topnav ul.dropdown-menu li:nth-of-type(5)"));
        Leads.click();
        WebDriverWait wait = new WebDriverWait (driver, Duration.ofSeconds(20));
        String text = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@class='moduleTitle']/h2"))).getText();
        Assert.assertEquals(" LEADS", text);
        // Find the table on the page and print the first 10 values in the Name column and the User column of the table to the console
        List<WebElement> nameColumn = driver.findElements(By.xpath("//table[contains(@class,'table-responsive')]/tbody/tr/td[3]"));
        System.out.println("Name Column values: ");
        int counter = 1;
        for (WebElement cell: nameColumn){
            System.out.println(cell.getText());
            counter+=1;
            if (counter==11){
                break;
            }
        }
        int counter1 = 1;
        List<WebElement> userColumn = driver.findElements(By.xpath("//table[contains(@class,'table-responsive')]/tbody/tr/td[8]"));
        System.out.println("User Column values: ");
        for (WebElement cell: userColumn){
            System.out.println(cell.getText());
            counter1+=1;
            if (counter1==11){
                break;
            }
        }
        // Close the browser
        driver.quit();
    }
}

File: Selenium/Project/PActivity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support import expected_conditions as EC

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the browser to the URL
    driver.get("http://alchemy.hguy.co/crm")
    # Find and select the username and password fields and enter the credentials
    driver.find_element(By.NAME, "user_name").send_keys("admin")
    driver.find_element(By.NAME, "username_password").send_keys("pa$$w0rd")
    # Click login
    driver.find_element(By.ID, "bigbutton").click()
    # Navigate to Sales -> Leads
    driver.find_element(By.ID, "grouptab_0").click()
    Leads = driver.find_element(By.CSS_SELECTOR, "ul.navbar-nav li.topnav ul.dropdown-menu li:nth-of-type(5)")
    Leads.click()
    wait = WebDriverWait(driver, 10)
    text = wait.until(EC.visibility_of_element_located((By.XPATH, "//*[@class='moduleTitle']/h2"))).text
    assert " LEADS" == text
    # Find the table on the page and print the first 10 values in the Name column and the User column of the table to
    # the console
    nameColumn = driver.find_elements(By.XPATH, "//table[contains(@class,'table-responsive')]/tbody/tr/td[3]")
    print("Name Column values: ")
    counter = 1
    for cell in nameColumn:
        print(cell.text)
        counter += 1
        if counter == 11:
            break
    counter1 = 1
    userColumn = driver.find_elements(By.XPATH, "//table[contains(@class,'table-responsive')]/tbody/tr/td[8]")
    print("User Column values: ")
    for cell in userColumn:
        print(cell.text)
        counter1 += 1
        if counter1 == 11:
            break
